### PR TITLE
SDIT-3720 Court movement inserted event - happy path

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementService.kt [hunk 1]
```diff
@@ -28,14 +49,158 @@ class CourtSchedulerSyncMovementService(
...
+      "directionCode" to directionCode,
+    )
+
+    if (event.originatesInDps) {
+      telemetryClient.trackEvent("${TELEMETRY_PREFIX}-inserted-skipped", telemetry)
+      return
+    }
+
+    mappingApi.getCourtMovementMappingOrNull(bookingId, movementSeq)
+      ?.also {
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementService.kt:66]: The main change is implementing this method - fairly standard sync to DPS stuff. Handles both OUT and IN court movements.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerMovementIntTest.kt [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-import kotlinx.coroutines.test.runTest
-import org.junit.jupiter.api.DisplayName
-import org.junit.jupiter.api.Nested
-import org.junit.jupiter.api.Test
-import org.mockito.kotlin.any
-import org.mockito.kotlin.verify
-import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.SqsIntegrationTestBase
-import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.sendMessage
-
-class CourtSchedulerMovementIntTest : SqsIntegrationTestBase() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerMovementIntTest.kt:12]: The test class was just renamed, not deleted.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementIntTest.kt [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+    reset(telemetryClient)
+  }
+
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  @DisplayName("EXTERNAL_MOVEMENT-CHANGED (OUT, inserted)")
+  inner class CourtMovementOutCreated {
+
+    @Nested
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementIntTest.kt:60]: Some tests for insert court movement OUT happy path.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementIntTest.kt [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+      verify(courtSchedulerSyncMovementService).courtMovementDeleted(any())
+    }
+  }
+
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  @DisplayName("EXTERNAL_MOVEMENT-CHANGED (IN, inserted)")
+  inner class CourtMovementInCreated {
+
+    @Nested
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/court/CourtSchedulerSyncMovementIntTest.kt:181]: Some tests for insert court movement IN happy path. Edge case tests coming next.

**Review by petergphillips — APPROVED** [review]
LGTM
